scripts/001-binutils-PPU.sh

```diff
@@ -1,40 +1,22 @@
 #!/bin/sh -e
 # binutils-PPU.sh by Naomi Peori (naomi@peori.ca)
 
-BINUTILS="binutils-2.22"
+# Set up our common variables and functions
+. $(git rev-parse --show-toplevel)/scripts/.common.sh
```

Comment:
Definitiely `dirname $0` here, it's actually in the same dir...

Reply:
dirname $0 is not always correct, there are multiple ways to call the script where $0 could be the script that called it instead of this script.

Reply:
If it's called correctly (i.e. not using "." or "source" or some such bullshit) then $0 will be correct.  Otherwise the shell would not have been able to open the script...

Reply:
dirname $0 is different for example when calling toolchain.sh vs scripts/001-binutils.sh for example.

Reply:
Yes, they are different strings, but point to the same directory.
When you run `toolchain.sh`, it cd:s into `build` and then runs `../scripts/001-binutils.sh`.  So `dirname $0` will yield `../scripts/`, and that is where the process will find `.common.sh` (before cd:ing anywhere else).
When you run `scripts/001-binutils.sh`, then `dirname $0` will yield `scripts/`, and again this is were to find `.common.sh`.
If you want a path that is still valid after changing your cwd, then you additinally need to call `realpath`, but for the purpose of including `.common.sh` that is not neeeded.